### PR TITLE
[now-node][now-next] Skip installing user dependencies for `now dev`

### LINKED ISSUE #2605: Functions in `now dev` waste 3 seconds on re-installing
As you can see in the video below, we are spending about 3 seconds on installing dependencies every time you edit a function - we need to resolve this differently:

[slow_function_rebuilds_-_Screen_Capture_7-22-19_4.15.02_PM.mp4.zip](https://github.com/zeit/now-cli/files/3426531/slow_function_rebuilds_-_Screen_Capture_7-22-19_4.15.02_PM.mp4.zip)

### LINKED ISSUE #2792: Stop installing user dependencies
As per @rauchg's suggestion, we've decided to publish a major update that **entirely** disables running `yarn` for the user.

---

This will drastically speed up `now dev` in general.

### PR DESCRIPTION
**Pull request by AndyBitz**
Skip installing user dependencies for `now dev` to speed up the processes.

This further fixes linting issues for `@now/next`.

Fixes #2792
Fixes #2605

### CODE REVIEW

**Review by Timer — DISMISSED** [review]
Looks like ES5 trailing commas were inadvertently turned off.

**Comment by styfle** [comment]
@Timer This was turned off in #2914 to unify the formatting in this monorepo. The point of the monorepo is to move common stuff to the top and formatting is one of those things. This matches the `now-cli` settings.

File: packages/now-next/tsconfig.json [hunk 1]
```diff
@@ -7,7 +7,9 @@
     "module": "commonjs",
     "outDir": "dist",
     "sourceMap": false,
-    "declaration": false
+    "declaration": false,
+    "allowSyntheticDefaultImports": true,
```
Comment (styfle) [inline packages/now-next/tsconfig.json]: I don't understand why `allowSyntheticDefaultImports` was enabled. That should be the same as `esModuleInterop` which is already enabled.

File: packages/now-next/package.json [hunk 2]
```diff
@@ -28,11 +28,12 @@
     "@zeit/node-file-trace": "0.2.13",
     "async-sema": "3.0.1",
     "buffer-crc32": "0.2.13",
+    "execa": "2.0.4",
     "fs-extra": "7.0.0",
     "get-port": "5.0.0",
     "resolve-from": "5.0.0",
     "semver": "6.1.1",
-    "typescript": "3.5.2",
+    "typescript": "3.6.2",
```
Comment (styfle) [inline packages/now-next/package.json]: Why bump the version of TS? We should do this for all packages at once.

**Review by styfle — COMMENTED** [review]
I originally implemented this here: https://github.com/zeit/now-builders/pull/812

I think we should have @TooTallNate look at it since my solution was similar but not very complete like this implementation.

File: packages/now-next/@types/zeit_node-file-trace/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
+declare module '@zeit/node-file-trace' {
```
Comment (styfle) [inline packages/now-next/@types/zeit_node-file-trace/index.d.ts]: Why was this added? It is already defined here: https://github.com/zeit/node-file-trace/blob/master/node-file-trace.d.ts

Comment (Timer) [reply packages/now-next/@types/zeit_node-file-trace/index.d.ts]: Because it's incomplete / not correct.

Comment (styfle) [reply packages/now-next/@types/zeit_node-file-trace/index.d.ts]: Then please update `node-file-trace` instead of creating workarounds here

File: packages/now-next/tsconfig.json (hunk 1 above)
Comment (AndyBitz) [reply packages/now-next/tsconfig.json]: Yea, I've ran into some issues and thought that would do it, but I'll remove it again

File: packages/now-next/package.json (hunk 2 above)
Comment (AndyBitz) [reply packages/now-next/package.json]: Reverted it

File: packages/now-next/package.json [hunk 4]
```diff
@@ -25,7 +25,7 @@
     "@types/resolve-from": "5.0.1",
     "@types/semver": "6.0.0",
     "@types/yazl": "2.4.1",
-    "@zeit/node-file-trace": "0.2.13",
+    "@zeit/node-file-trace": "0.2.14",
```
Comment (styfle) [inline packages/now-next/package.json:28]: Can you update `now-node` as well so these two can use the same version?

File: .prettierrc [hunk 5]
```diff
@@ -1,3 +1,4 @@
 {
-  "singleQuote": true
-}
\ No newline at end of file
+  "singleQuote": true,
+  "trailingCommas": "es5"
```
Comment (styfle) [inline .prettierrc:3]: @lucleray Is this sufficient or does something need to be changed in eslint config?

Comment (lucleray) [reply .prettierrc:3]: This is perfect 👍 Eslint rules that overlap with prettier are deactivated by `eslint-config-prettier` here:
https://github.com/zeit/now/blob/cc0bd9f0a226232c4919c14183c73b8138c6c238/.eslintrc.json#L13

**Review by styfle — APPROVED** [review]
I'm fine with this solution. One thing to consider is that any docs that say `git clone repo && now dev` will need to include `npm install` or `yarn install` before `now dev`.

cc @msweeneydev @timothyis
